lib/semian/circuit_breaker.rb

```diff
@@ -0,0 +1,131 @@
+module Semian
+  class CircuitBreaker
+    extend Forwardable
+    attr_reader :state
+
+    def_delegators :@resource, :acquire, :destroy, :count, :semid
```

Comment:
I see you point. I did this to not change the current Semian API, but since I bout many people are using it yet, we can find a better way. Let's chat a bit.